Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM rust:alpine AS builder
+
+RUN apk add \
+    build-base \
+    openssl-dev
+
+WORKDIR /usr/src/ferris-bot
+COPY assets assets
+COPY src src
+COPY Cargo.toml .
+RUN cargo build --release
+
+FROM alpine
+COPY --from=builder /usr/src/ferris-bot/target/release/twitch_queue_bot /usr/bin
+ENTRYPOINT \
+    /usr/bin/twitch_queue_bot \
+    --config-file /config/ferrisbot.toml \
+    --first-token-file /config/first-token.json
```

Comment:
one problem is that --first-token-file (which I plan to remove as part of #29 ) is only needed on first run, and it's actually harmful on every subsequent run. is it possible, for now, to pass it as an extra parameter on first launch in some way?

Reply:
The container is usually only used once, so there would not be a second run.
it is possible to remove it from the entrypoint and pass it manually when starting the container tho, but im not sure if that is better.

Reply:
I'm thinking of cases in which e.g. a VM automatically starts this docker on bootup, and the VM is rebooted. Are you saying that we would lose all state in that case? (the token is refreshed periodically and stored in a local file)

Reply:
Any persistent data should be stored in a docker volume, otherwise it is lost. The only volume in the example `docker-compose.yml` is for config files. If any more data should be kept, it would have to be mounted via another volume, or ideally be included in the `/config` directory. Besides that, having to use a different command on the first run and not on subsequent runs is pretty strange. We should consider ignoring the argument if it is not needed, or using another solution.

Reply:
I agree about the oddity of the first-time only argument, we can probably ignore that for now and find a way to bootstrap the token process before composing the docker. We can address this in a separate PR if you prefer, but the idea would be (for now) to run the bot ones and then copy the token stored in the path under `twitch.token_filepath`.

But we do need to provide an example of how to configure this so that the refreshed token is persisted. The refreshed token is checkpointed in whatever file is indicated  in the `twitch.token_filepath` config param. How can we make sure that a docker volume is mounted and persisted? we can then recommend changing the config param to have the right path.

Reply:
We could recommend to set `twitch.token_filepath` to a file located in `/config` so it would be in the docker volume. As a workaround for the first-time argument, we could include a script in the container which could check if the file where the token is saved exists, and if so doesn't supply the argument

Reply:
sgtm

Reply:
So how would we check if the bot saved the tokens? It would ideally have to be a file at some known location, as otherwise the script couldn't check for it.

Reply:
I guess for now we can just mandate what the file name should be in `twitch.token_filepath`? given that we're going to effectively mandate using `/config` in the path, we might as well mandate the whole filename for the Docker case.

Reply:
That is true, but the problem is that we cannot assure the value to be set accordingly. Maybe we should wait with this until #29 is addressed?

Reply:
as discussed in chat, we have a generic problem of handling tokens, and solving #29 isn't necessarily going to solve that. should we have a first version of docker in which we need to place the token somewhere (so it needs to be obtained before creating the token) and then we can design a better keystore-like solution?

Reply:
#29 is addressed.

I think the best option we have at the moment is to instruct the user to obtain auth credentials by running locally, and then copy the token inside the docker, adjusting the `token_filepath` parameter in their `ferrisbot.toml`